src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implementation of health heartbeat functionality.
+    /// </summary>
+    internal class HealthHeartbeatProvider : IDisposable, IHeartbeatProvider
+    {
+        /// <summary>
+        /// The default interval between heartbeats if not specified by the user
+        /// </summary>
+        public static int DefaultHeartbeatIntervalMs = 5000;
+
+        /// <summary>
+        /// The default fields to include in every heartbeat sent. Note that setting the value to '*' includes all default fields.
+        /// </summary>
+        public static string DefaultAllowedFieldsInHeartbeatPayload = "*";
+
+        /// <summary>
+        /// The name of the health heartbeat metric item and operation context.
+        /// </summary>
+        public static string HeartbeatSyntheticMetricName = "SDKHeartbeat";
+
+        /// <summary>
+        /// The payload items to send out with each health heartbeat.
+        /// </summary>
+        private Dictionary<string, IHealthHeartbeatPayloadExtension> payloadItems;
+        
+        private bool disposedValue = false; // To detect redundant calls to dispose
+        private int intervalBetweenHeartbeatsMs; // time between heartbeats emitted specified in milliseconds
+        private string enabledHeartbeatPayloadFields; // string containing fields that are enabled in the payload. * means everything available.
+        private TelemetryClient telemetryClient; // client to use in sending our heartbeat
+        private int heartbeatsSent;
```

Comment:
`unit`?

Reply:
Unit of which item? Sorry - cannot see which field you are referring to here... heartbeatsSent perhaps can be renamed to heartbeatsSentCount...